src/js/components/Courses/Courses.js

```diff
@@ -28,17 +31,34 @@ class Courses extends PureComponent {
       .catch(() => this.setState({ courses: [] }));
   }
 
+  handleClickMore = () => {
+    this.setState(prevState => ({ page: prevState.page + 1 }), () => {
+      getMoreData(this.state.page, limit)
+        .then((courses) => {
+          this.setState({ loadedDataLength: courses.length });
+          this.setState(prevState => ({ courses: [...prevState.courses, ...courses] }));
+        })
+        .then(() => this.setState({ isFetching: false }))
+        .catch(() => this.setState({ page: 1 }));
+    });
+  }
+
+  isAllDataLoaded = () => this.state.loadedDataLength < limit
```

Comment:
could you explain this logic to me, i don't get it

Reply:
when we click on Load more button, we expect that we will see a few new items (f.e. 4 items if limit=4). loadedDataLength shows us how many items came from server when we clicked on LoadMore button.  And if (loadedDataLength < limit) it means that this is the last portion of data that server could give us

Reply:
then why it < not <= ?)

Reply:
because if limit=4 (f.e.) then server will return us 4 new items on loadMore click. If we will use <= then isAllDataLoaded function will return true constantly, although it's not right
If server has 10 items summary, then on the first loadMore click, server give us 4 items, isAllDataLoaded will return false (as 4 not less then 4). On the second loadMore click server will give 4 items again, isAllDataLoaded will return false again. On the third loadMore click server will give us 2 rest items. isAllDataLoaded will return true (because 2<4 it's true) and this means that this was last portion of data